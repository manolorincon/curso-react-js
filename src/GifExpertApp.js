import React, { Fragment, useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Dragon Ball Z'])

    // const handleAdds = () => {
    //     setCategories( array => [...array, 'Power Rangers']);
    //     //tambien podr√≠amos usar:
    //     //setCategories( [...categories, 'Power Rangers']);
    // }

    return (
        <Fragment>
            <h1>GifExpertApp</h1>
            <AddCategory
                categories = { categories }
                setCategories = { setCategories }
            />
            <hr />

            <ol>
                {
                    categories.map( category => (
                        <GifGrid
                            key = { category } 
                            category = { category } 
                        />
                    ))
                }
            </ol>

        </Fragment>
     );
}
 
export default GifExpertApp;