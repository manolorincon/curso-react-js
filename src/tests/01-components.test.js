import React from 'react';
import GifExpertApp from '../GifExpertApp';
import {shallow} from 'enzyme';

describe('Primer test para componentes', () => {

    test('Debe retornar OK', () => {

        const wrapper = shallow(<GifExpertApp />)
        
        expect(wrapper).toMatchSnapshot();
    })
    
    
})
