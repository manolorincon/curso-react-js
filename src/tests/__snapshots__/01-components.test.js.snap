// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primer test para componentes Debe retornar OK 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    categories={
      Array [
        "Dragon Ball Z",
      ]
    }
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon Ball Z"
      key="Dragon Ball Z"
    />
  </ol>
</Fragment>
`;
